src/win32.c

```diff
@@ -3,6 +3,54 @@
 
 #include "common.h"
 
+#if defined(USE_LIBXMP_ASPRINTF)
+
+#undef asprintf
+
+int libxmp_asprintf(char **strp, const char *fmt, ...)
+{
+	int n;
+	int size = 100;     /* Guess we need no more than 100 bytes */
+	char *p, *np;
+	va_list ap;
+
+	*strp = NULL;
+
+	if ((p = malloc(size)) == NULL)
+		return -1;
+
+	while (1) {
+		/* Try to print in the allocated space */
+		va_start(ap, fmt);
+		n = vsnprintf(p, size, fmt, ap);
+		va_end(ap);
```

Comment:
While we're here, this seems suspicious—my understanding, which may be wrong, is that you're allowed to call `va_start` on `fmt` exactly one time, and you are required to use `va_copy` if you need to make multiple passes.

https://stackoverflow.com/questions/5339653/segmentation-fault-in-va-args-parsing